text/0063-doc-comments.md

```diff
@@ -0,0 +1,226 @@
+            { "type": "Action", "id": "CreateList" },
+            { "type": "Action", "id": "GetList" }
+        ]
+    },
+    "resource": {
+        "op": "==",
+        "entity": { "type": "Application", "id": "TinyTodo" }
+    }
+}
+```
+
+and in a schema:
+
+```json
+{
+    "entityTypes": {
+        "User": {
+            "doc": "This entity defines our central user type",
+            "shape": {
+                "type": "Record",
+                "attributes": {
+                    "location": { 
+                        "doc": "The location where this user works. E.g., ABC17 or XYZ77.",
+                        "type": "String"
+                    },
+                    "jobLevel": {
+                        "doc": "A numeric level describing this users job.",
+                        "type": "Long"
+                    },
+                }
+            }
+        }   
+    }
+}
+```
+
+## Drawbacks
+
+1. This is a significant change to the parsers for policies and schema, both in their human readable and JSON formats.
+   There is always a possibility of mistakes when making changes to our parsers, especially given that they are not modeled and differentially tested.
+2. This will eventually be a breaking change to the policy format.
+   Although, as noted above, we don't need to make this change immediately, and we could even opt to never make the change.
+3. We are likely to find that users want to add doc comments in a position we didn't anticipate. We will be able to introduce doc comments in more positions in the future without a breaking change, so argument over where else we should support doc comments should ideally not delay acceptance of this RFC if we are confident that the currently proposed positions should be supported.
+4. It partially duplicates the functionality of annotations, particularly given that we can use multi-line strings as annotation values.
+
+## Alternatives
+
+### Regular comments are good enough
+
+Our position would be that documentation should be written in standard Cedar comments. 
+If someone wants to attach a particular meaning to and write documentation tools around comments starting with `///`, they are free to do so, but we do not explicitly support this.
+This requires no changes to Cedar, and is maximally flexible (standard comments can appear anywhere).
+
+Asking users to develop their own documentation tools could resulted in multiple ad-hoc tools that are likely to fail to handle certain edge cases (e.g., `///` in a string) unless they substantially re-implement the Cedar parser.
+Alternatively, we could provide the tools for generating documentation in this manner, but this implies standardizing the format and position of these comments, which is essentially what this RFC proposes, just without considering JSON formats.
+
+Comments do not exist in the JSON format, so this alternative provides nothing for documenting a JSON formatted policy or schema.
+
+### Annotations as documentation
```

Comment:
This is an interesting option to me, but I still prefer the look of `///` (or whatever syntax is chosen). I think it's fine to allow some overlap in functionality between annotations and doc comments. It might also be useful to unify their implementation in the future so that `///` is syntactic sugar for `@doc(...)`.

Reply:
Yes. The doc comment syntax being syntactic sugar under the hood is something I had in mind while writing this. It requires a substantial extension to extensions beyond ever RFC #48, but perhaps people will find other application of generalized annotations.